@commitlint/cli/src/cli.ts

```
@@ -384,6 +387,9 @@ async function main(args: MainArgs): Promise<void> {
 			throw new CliError(output, pkg.name, 2);
 		}
 	}
+	if (!report.valid && isRulesEmpty) {
```

@escapedcat if isRulesEmpty=true at line 347, do we really need to perform the invokation at line 348 or can return early?

---

Tried that before. We need a "formatted" `report` (aka `output`) to display the current message.  
We could just throw a new error but I kinda like the current formatted way.

---

still, wouldn't it be better to check `isRulesEmpty` first? i.e. `if (isRulesEmpty && (!report.valid)) {`, micro-optimization there; or even just `if (isRulesEmpty) {`

---

Got it, adjusted, thanks!